runbotics/runbotics-orchestrator-ui/src/src-app/components/cron/Cron.tsx

```diff
@@ -1,47 +1,47 @@
 
-import { Button } from '@mui/material';
-
+import If from '../utils/If';
 import { setValuesFromCronString, getCronStringFromValues } from './converter';
-import Hours from './fields/Hours';
-import Minutes from './fields/Minutes';
-import MonthDays from './fields/MonthDays';
-import Months from './fields/Months';
+import { ClearButton } from './Cron.styles';
+import AllMonthDays from './fields/AllMonthDays/';
+import AllWeekDays from './fields/AllWeekDays/';
+import Hours from './fields/Hours/';
+import Minutes from './fields/Minutes/';
+import Months from './fields/Months/';
 import Period from './fields/Period';
-import WeekDays from './fields/WeekDays';
 import DEFAULT_LOCALE_EN from './locale';
-import { CronProps, PeriodType } from './types';
+import { CronProps, PeriodType, PeriodTypes } from './types';
 import { classNames, setError, usePrevious } from './utils';
 
 // eslint-disable-next-line max-lines-per-function
-export default function Cron(props: CronProps) {
-    const {
-        clearButton = true,
-        clearButtonProps = {},
-        clearButtonAction = 'fill-with-every',
-        locale = DEFAULT_LOCALE_EN,
-        value = '',
-        setValue,
-        displayError = true,
-        onError,
-        className,
-        defaultPeriod = 'day',
-        allowEmpty = 'for-default-value',
-        humanizeLabels = true,
-        humanizeValue = false,
-        disabled = false,
-        readOnly = false,
-        leadingZero = false,
-        shortcuts = ['@yearly', '@annually', '@monthly', '@weekly', '@daily', '@midnight', '@hourly'],
-        clockFormat,
-    } = props;
+export default function Cron({
+    clearButton = true,
+    clearButtonAction = 'fill-with-every',
+    locale = DEFAULT_LOCALE_EN,
+    value = '',
+    setValue,
+    displayError = true,
+    onError,
+    className,
+    defaultPeriod = PeriodTypes.DAY,
+    allowEmpty = 'for-default-value',
+    humanizeLabels = true,
+    humanizeValue = false,
+    disabled = false,
+    readOnly = false,
+    leadingZero = false,
+    shortcuts = ['@yearly', '@annually', '@monthly', '@weekly', '@daily', '@midnight', '@hourly'],
+    clockFormat,
+}: CronProps) {
     const internalValueRef = useRef<string>(value);
     const defaultPeriodRef = useRef<PeriodType>(defaultPeriod);
-    const [period, setPeriod] = useState<PeriodType | string | undefined>();
+    const [period, setPeriod] = useState<PeriodType | undefined>();
     const [monthDays, setMonthDays] = useState<number[] | undefined>();
+    const [nthMonthDays, setNthMonthDays] = useState<number[] | undefined>();
     const [months, setMonths] = useState<number[] | undefined>();
     const [weekDays, setWeekDays] = useState<number[] | undefined>();
+    const [nthWeekDays, setNthWeekDays] = useState<number[] | undefined>();
     const [hours, setHours] = useState<number[] | undefined>();
     const [minutes, setMinutes] = useState<number[] | undefined>();
```

Comment:
in my humble opinion this should be one state object/useReducer. It's getting really messy

Reply:
done sir